packages/node_modules/@node-red/editor-client/src/js/ui/deploy.js

```diff
@@ -450,21 +512,17 @@ RED.deploy = (function() {
         const nns = RED.nodes.createCompleteNodeSet();
         const startTime = Date.now();
 
-        $(".red-ui-deploy-button-content").css('opacity', 0);
-        $(".red-ui-deploy-button-spinner").show();
-        $("#red-ui-header-button-deploy").addClass("disabled");
-
+        deployButtonSetBusy();
         const data = { flows: nns };
-
-        if (!force) {
+        data.runtimeState = RED.runtime.state;
+        if (data.runtimeState === RED.runtime.states.STOPPED || force) {
+            data._rev = RED.nodes.version();
```

Comment:
Not seen this mentioned anywhere in the design - it appears you're also modify the payload of the `/flows` endpoint?

Can you explain the purpose of including `runtimeState` and `_rev` in the request?

Reply:
This is to ensure that the state of the editor and the runtime are in agreement (a simple sanity check)

Reply:
But why is it needed? We already use the `rev` property to check the editor is deploying on top of the latest set of flows. I don't see where the need to include the state of the editor and the `_rev` property helps.